src/kup/package.py

```diff
@@ -1,14 +1,34 @@
 from typing import Mapping, Optional, Union
 
 
+class PackageName:
+    __slots__ = ['base', 'ext']
+
+    def __init__(
+        self,
+        base: str,
+        ext: Optional[list[str]] = None,
+    ):
+        self.base = base
+        self.ext = ext if ext is not None else []
```

Comment:
This could be turned into a dataclass, then you can compare instances for equality, put them in a set, etc...

```suggestion
dataclass(frozen=True)
class PackageName:
    base: str
    ext: tuple[str, ...]

    def __init__(self, base: str, ext: Iterable[str] = ()):
        object.__setattr__(self, 'base', base)
        object.__setattr__(self, 'ext', tuple(ext))
```